Simplified/NYPLUserAccount.swift

```diff
@@ -0,0 +1,402 @@
+      return
+    }
+    
+    NYPLUserAccount.accountInfoLock.lock()
+    defer {
+        NYPLUserAccount.accountInfoLock.unlock()
+    }
+    
+    sharedKeychain.setObject(patron, forKey: patronKey)
+    
+    NotificationCenter.default.post(
+      name: Notification.Name.NYPLUserAccountDidChange,
+      object: self
+    )
+  }
+  
+  @objc(setAuthToken:)
+  func setAuthToken(token: String) {
+    guard let sharedKeychain = NYPLKeychain.shared() else {
+      return
+    }
+    
+    NYPLUserAccount.accountInfoLock.lock()
+    defer {
+        NYPLUserAccount.accountInfoLock.unlock()
+    }
+    
+    sharedKeychain.setObject(token, forKey: authTokenKey)
+    
+    NotificationCenter.default.post(
+      name: Notification.Name.NYPLUserAccountDidChange,
+      object: self
+    )
+  }
+  
+  @objc(setProvider:)
+  func setProvider(provider: String) {
+    guard let sharedKeychain = NYPLKeychain.shared() else {
+      return
+    }
+    
+    NYPLUserAccount.accountInfoLock.lock()
+    defer {
+        NYPLUserAccount.accountInfoLock.unlock()
+    }
+    
+    sharedKeychain.setObject(provider, forKey: providerKey)
+    
+    NotificationCenter.default.post(
+      name: Notification.Name.NYPLUserAccountDidChange,
+      object: self
+    )
+  }
+  
+  @objc(setUserID:)
+  func setUserID(id: String) {
+    guard let sharedKeychain = NYPLKeychain.shared() else {
+      return
+    }
+    
+    NYPLUserAccount.accountInfoLock.lock()
+    defer {
+        NYPLUserAccount.accountInfoLock.unlock()
+    }
+    
+    sharedKeychain.setObject(id, forKey: userIDKey)
+    
+    NotificationCenter.default.post(
+      name: Notification.Name.NYPLUserAccountDidChange,
+      object: self
+    )
+  }
+  
+  @objc(setDeviceID:)
+  func setDeviceID(id: String) {
+    guard let sharedKeychain = NYPLKeychain.shared() else {
+      return
+    }
+    
+    NYPLUserAccount.accountInfoLock.lock()
+    defer {
+        NYPLUserAccount.accountInfoLock.unlock()
+    }
+    
+    sharedKeychain.setObject(id, forKey: deviceIDKey)
+    
+    NotificationCenter.default.post(
+      name: Notification.Name.NYPLUserAccountDidChange,
+      object: self
+    )
+  }
+    
+  // MARK: - Remove
+    
+  func removeBarcodeAndPIN() {
+    guard let sharedKeychain = NYPLKeychain.shared() else {
+      return
+    }
+    
+    NYPLUserAccount.accountInfoLock.lock()
+    defer {
+        NYPLUserAccount.accountInfoLock.unlock()
+    }
+    
+    sharedKeychain.removeObject(forKey: barcodeKey)
+    sharedKeychain.removeObject(forKey: authorizationIdentifierKey)
+    sharedKeychain.removeObject(forKey: PINKey)
+    
+    NotificationCenter.default.post(
+      name: Notification.Name.NYPLUserAccountDidChange,
+      object: self
+    )
+  }
+  
+  func removeAll() {
+    guard let sharedKeychain = NYPLKeychain.shared() else {
+      return
+    }
+    
+    NYPLUserAccount.accountInfoLock.lock()
+    defer {
+        NYPLUserAccount.accountInfoLock.unlock()
```

Comment:
where you able to test this code at runtime? this `removeAll` function will unlock the lock twice. Not sure if that's undefined behavior. You can also enable the undefined behavior sanitizer in the scheme run diagnostics settings to verify.